utils/network.go

```diff
@@ -6,151 +6,229 @@ import (
+		baseURL:             "https://alpha-mainnet.starknet.io/",
+		chainID:             "SN_MAIN",
+		l1ChainID:           big.NewInt(1),
+		coreContractAddress: common.HexToAddress("0xc662c410C0ECf747543f5bA90660f6ABeBD9C8c4"),
+		blockHashMetaInfo:   &MetaInfoMAINNET,
+	}
+	GOERLI = Network{
+		name:    "goerli",
+		baseURL: "https://alpha4.starknet.io/",
+		chainID: "SN_GOERLI",
+		//nolint:gomnd
+		l1ChainID:           big.NewInt(5),
+		coreContractAddress: common.HexToAddress("0xde29d060D45901Fb19ED6C6e959EB22d8626708e"),
+		blockHashMetaInfo:   &MetaInfoGOERLI,
+	}
+	GOERLI2 = Network{
+		name:    "goerli2",
+		baseURL: "https://alpha4-2.starknet.io/",
+		chainID: "SN_GOERLI2",
+		//nolint:gomnd
+		l1ChainID:           big.NewInt(5),
+		coreContractAddress: common.HexToAddress("0xa4eD3aD27c294565cB0DCc993BDdCC75432D498c"),
+		blockHashMetaInfo:   &MetaInfoGOERLI2,
+	}
+	INTEGRATION = Network{
+		name:              "integration",
+		baseURL:           "https://external.integration.starknet.io/",
+		chainID:           "SN_GOERLI",
+		blockHashMetaInfo: &MetaInfoINTEGRATION,
+	}
+	MetaInfoMAINNET = blockHashMetaInfo{
+		First07Block:             833,
+		FallBackSequencerAddress: fallBackSequencerAddress,
+	}
+	MetaInfoGOERLI = blockHashMetaInfo{
+		First07Block:             47028,
+		UnverifiableRange:        []uint64{119802, 148428},
+		FallBackSequencerAddress: fallBackSequencerAddress,
+	}
+	MetaInfoGOERLI2 = blockHashMetaInfo{
+		First07Block:             0,
+		FallBackSequencerAddress: fallBackSequencerAddress,
+	}
+	MetaInfoINTEGRATION = blockHashMetaInfo{
+		First07Block:             110511,
+		UnverifiableRange:        []uint64{0, 110511},
+		FallBackSequencerAddress: fallBackSequencerAddress,
+	}
 )
 
 func (n Network) String() string {
-	switch n {
-	case MAINNET:
-		return "mainnet"
-	case GOERLI:
-		return "goerli"
-	case GOERLI2:
-		return "goerli2"
-	case INTEGRATION:
-		return "integration"
-	default:
-		// Should not happen.
-		panic(ErrUnknownNetwork)
-	}
+	return n.name
 }
 
 func (n Network) MarshalYAML() (interface{}, error) {
 	return n.String(), nil
 }
 
-func (n *Network) MarshalJSON() ([]byte, error) {
+func (n Network) MarshalJSON() ([]byte, error) {
 	return json.RawMessage(`"` + n.String() + `"`), nil
 }
 
 func (n *Network) Set(s string) error {
-	switch s {
-	case "MAINNET", "mainnet":
-		*n = MAINNET
-	case "GOERLI", "goerli":
-		*n = GOERLI
-	case "GOERLI2", "goerli2":
-		*n = GOERLI2
-	case "INTEGRATION", "integration":
-		*n = INTEGRATION
-	default:
+	predefinedNetworks := map[string]Network{
+		"MAINNET":     MAINNET,
+		"mainnet":     MAINNET,
+		"GOERLI":      GOERLI,
+		"goerli":      GOERLI,
+		"GOERLI2":     GOERLI2,
+		"goerli2":     GOERLI2,
+		"INTEGRATION": INTEGRATION,
+		"integration": INTEGRATION,
+	}
+
+	if network, ok := predefinedNetworks[s]; ok {
+		*n = network
+		return nil
+	}
+
+	return n.setCustomNetwork(s)
+}
+
+func (n *Network) setCustomNetwork(s string) error {
```

Comment:
I'm not sure how well this will scale. In the future, we will likely need a lot more information for a custom network, e.g., the genesis state and hard fork history. Can we make this accept a json string and unmarshal it to the `Network` struct? That should be easier than manually parsing it.

Just a thought, I'm open to other ideas.

Reply:
I implemented the json-string approach